### PR TITLE
Add clipboard functionality to platform plugin

### PR DESCRIPTION
**Pull request by abarth**
This will eventually replace the clipboard mojom.

### CODE REVIEW

**Comment by abarth** [comment]
@chinmaygarde @jason-simmons

File: shell/platform/android/io/flutter/plugin/platform/PlatformPlugin.java [hunk 1]
```diff
@@ -166,6 +175,28 @@ private void popSystemNavigator() {
...
+              result.put("text", clip.getItemAt(0).getText().toString());
+              return result;
+        }
+
+        return null;
+    }
+
+    private void setClipboardData(JSONObject data) throws JSONException {
+        ClipboardManager clipboard = (ClipboardManager) mActivity.getSystemService(Context.CLIPBOARD_SERVICE);
+        ClipData clip = ClipData.newPlainText("text label?", data.getString("text"));
```
Comment (chinmaygarde) [inline shell/platform/android/io/flutter/plugin/platform/PlatformPlugin.java]: Maybe just have `data.GetString("text")` as the user label as well. `"text label?"` does not seem very descriptive.

**Comment by jason-simmons** [comment]
LGTM

File: shell/platform/android/io/flutter/plugin/platform/PlatformPlugin.java [hunk 2]
```diff
@@ -166,6 +175,28 @@ private void popSystemNavigator() {
...
+    private JSONObject getClipboardData(String format) throws JSONException {
+        ClipboardManager clipboard = (ClipboardManager) mActivity.getSystemService(Context.CLIPBOARD_SERVICE);
+        ClipData clip = clipboard.getPrimaryClip();
+        if (clip == null)
+            return null;
+
+        if ((format == null || format.equals(kTextPlainFormat)) &&
+            clip.getDescription().hasMimeType(ClipDescription.MIMETYPE_TEXT_PLAIN)) {
+              JSONObject result = new JSONObject();
+              result.put("text", clip.getItemAt(0).getText().toString());
```
Comment (chinmaygarde) [inline shell/platform/android/io/flutter/plugin/platform/PlatformPlugin.java]: The mime type check only verifies that the clip data contains plain text. Are you certain the item at index 0 contains that plain text information? Must admit, not very familiar with the API.

File: shell/platform/android/io/flutter/plugin/platform/PlatformPlugin.java (hunk 1 above)
Comment (abarth) [inline shell/platform/android/io/flutter/plugin/platform/PlatformPlugin.java]: These are both cargo-culted from the existing code.  /cc @mpcomplete

**Comment by abarth** [comment]
I'm going to merge this patch because it contains the same code as the old version.  If we want to change the code, we should do that in a separate patch.

File: shell/platform/android/io/flutter/plugin/platform/PlatformPlugin.java (hunk 1 above)
Comment (mpcomplete) [inline shell/platform/android/io/flutter/plugin/platform/PlatformPlugin.java]: Yeah I wasn't very clear on the Android API. The "text label?" param seems arbitrary, not sure how it is used.

**Pull request merged**